test/arrays.js

```diff
@@ -137,6 +137,33 @@
     equal(_.sortedIndex(array, 2147483648), 2147483648, 'should work with large indexes');
   });
 
+  test('sortedIndex conforms to sortBy for undefined (#1834)', function() {
+    var sorted = _.sortBy([undefined, 1, undefined, 2]);
+    equal(_.sortedIndex(sorted, undefined, _.identity), 2);
+
+    var edgeCaseNumbers = [-Infinity, -Infinity, 0, 345345345234, Infinity, Infinity, undefined, undefined, NaN];
```

Comment:
I don't think it matters what order they go in, the most important thing is `_.sortBy` and `_.sortedIndex` are consistent with one another for the values